packages/peer-exchange/src/waku_peer_exchange_discovery.ts

```diff
@@ -109,6 +121,33 @@ export class PeerExchangeDiscovery
     return "@waku/peer-exchange";
   }
 
+  private readonly startRecurringQueries = async (
+    peerId: PeerId
+  ): Promise<void> => {
+    const peerIdStr = peerId.toString();
+    const {
+      queryInterval = DEFAULT_PEER_EXCHANGE_QUERY_INTERVAL_MS,
+      attemptsBeforeAbort = DEFAULT_ATTEMPTS_BEFORE_ABORT,
+    } = this.options;
+
+    await this.query(peerId);
+
+    const interval = setInterval(async () => {
+      const currentAttempt = this.queryAttempts.get(peerIdStr) ?? 1;
+
+      if (currentAttempt > attemptsBeforeAbort) {
+        this.abortQueriesForPeer(peerIdStr, interval);
```

Comment:
`interval` has not yet been initialized. Also, I would not use the same variable name than at L#135.

You probably need to get the `_interval` value from the `intervals` map first, and if set, then you can check the `currentAttempt`.